test/parallel/test-fs-write-file.js

```diff
@@ -95,3 +95,16 @@ fs.open(filename4, 'w+', common.mustSucceed((fd) => {
 
   process.nextTick(() => controller.abort());
 }
+
+{
+  // Test read-only mode
+  const filename = join(tmpdir.path, 'test6.txt');
+  fs.writeFileSync(filename, '');
+
+  const readOnlyOption = { mode: fs.constants.O_RDONLY, flag: 'r' };
```

Comment:
What you passed as `mode` is a `flag`. The `mode` could be any of these OR-ed together: `S_IRWXU`, `S_IRWXO`, ...

Reply:
Oh, thanks for the catch! In this case the `mode` seems not necessary. Updated